packages/react-component-library/src/components/DatePicker/useOpenClose.ts

```diff
@@ -13,14 +13,15 @@ function hasParentWithMatchingSelector(
 
 export function useOpenClose(
   ref: React.RefObject<undefined>,
-  isOpen?: boolean
+  selector: string,
+  isOpen?: boolean,
 ) {
   const [openState, setOpenState] = useState<boolean>(isOpen)
 
   useDocumentClick(ref, (event: Event) => {
     const target = event.target as Node
 
-    if (!hasParentWithMatchingSelector(target, '.rn-date-picker__container')) {
+    if (!hasParentWithMatchingSelector(target, selector)) {
```

Comment:
I am wondering if this is the opportunity to rework some of this as maybe there are other options rather than using the `selector`.

Reply:
I just had a look through this an i'm not sure what an alternative implementation would be. 

I just attempted to do something with refs but you need to work directly with the DOM because the react-tether library moves things out of the page slow and component hierarchy so that it can absolutely position them relative to the viewport.